dnf/base.py

```diff
@@ -1575,15 +1577,20 @@ def upgrade(self, pkg_spec, reponame=None):
         if any((s.matches() for s in sltrs)):
             prev_count = self._goal.req_length()
             installed = self.sack.query().installed()
+            available = self.sack.query().available()
             for sltr in sltrs:
                 if not sltr.matches():
                     continue
                 pkg_name = sltr.matches()[0].name
                 if not installed.filter(name=pkg_name):
                     # wildcard shouldn't print not installed packages
                     if not wildcard:
-                        msg = _("Package %s not installed, cannot update it.")
-                        logger.warning(msg, pkg_name)
+                        if available.filter(name=pkg_name):
+                            msg = _('Package %s%s%s available, but not installed.')
```

Comment:
I have an inspiration from yum - _('Package(s) %s%s%s available, but not installed.')
And from dnf - _('Installed package %s%s%s%s not available.').
It highlights the package in output.  If I use empty string in substitution, I only reuse same string that on other place produce highlighted output. The reason is that only one string will be allowed for translation.